node/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -89,75 +89,105 @@ retrievalWorkerImpl SendActions {..} =
         queue        <- view (lensOf @BlockRetrievalQueueTag)
         recHeaderVar <- view (lensOf @RecoveryHeaderTag)
         logDebug "Waiting on the block queue or recovery header var"
+        -- Reading the queue is a priority, because it sets the recovery
+        -- variable in case the header is classified as alternative. So if the
+        -- queue contains lots of headers after a long delay, we'll first
+        -- iterate over them and set recovery variable to the latest one, and
+        -- only then we'll do recovery.
```

Comment:
Where does this iteration happen? Comment should be placed closer to it.

Reply:
This comment may be in the middle of the iteration loop. Note `mainLoop` is called recursively.